common/db.go

```diff
@@ -323,6 +337,45 @@ func (db *DB) LastSeenBlock() (*big.Int, error) {
 	return header.Number, nil
 }
 
+func (db *DB) NetworkID() (*big.Int, error) {
+	idString, err := db.SelectKV("network")
+	if err != nil {
+		return nil, err
+	}
+
+	if idString == "" {
+		return nil, nil
+	}
+
+	id, ok := new(big.Int).SetString(idString, 10)
+	if !ok {
+		return nil, fmt.Errorf("unable to convert networkID string to big.Int")
+	}
+
+	return id, nil
+}
+
+func (db *DB) SelectKV(key string) (string, error) {
```

Comment:
Hm while I see why exporting these methods could be useful, I feel like this is leaking details of the DB to the caller (i.e. that we're using a KV table) and am wary of exposing it/encouraging storing/fetching KV pairs with the caller passing in string keys every time they need to call these methods - feels a bit fragile. Isolating the use of string keys to query the underlying KV table in this file feels safer IMO.

Reply:
Yep